### DISCUSSION TITLE
Skip some subpackage when building wheel

### DISCUSSION THREAD

**Discussion by viblo** [pyodide/pyodide#4206] [Q&A]
I have a library that I build a pyodide wasm wheel for. It has some parts that are not really useful in a wasm-setting, so I would like to skip them when building to make the resulting wheel file smaller. 

How to do this in the best way?

**Comment by rth** [comment]
Assuming you built the wheel via `pyodide build` (i.e. out of tree), I feel the easiest is to run [`wheel unpack`](https://wheel.readthedocs.io/en/stable/reference/wheel_unpack.html) remove the files you don't need and then run `wheel pack` to get back the wheel. This can be done with a script. 

We should maybe add this functionality to `pyodide build` via an exclude regexp or some other mechanism.

**Comment by ryanking13** [comment]
My guess is that this is less of an issue with Pyodide and more of an issue with Python packaging in general.

The way I think about it is to split that package into two packages, one for common use and one for platform-specific use.

**Comment by viblo** [comment]
Thanks for the suggestions. A third option I thought of after reading your suggestions was to add a little logic in setup.py to exclude the subpackages based on an environment variable, and then set this variable before calling `pyodide build`. (Requires setup.py, but Im required to use that anyway for CFFI).
